### PR TITLE
Don't create a promise on JSON.parse, avoid ugly onProcess callback

### PR DESCRIPTION
**Pull request by Pchelolo**
Follow-up on https://github.com/wikimedia/change-propagation/pull/101

- Some generic refactoring
- More sync processing until we verify that the message will indeed be processed - less promises, should save some perf/mem
- Added a stat on dequeue speed

cc @wikimedia/services

### CODE REVIEW

File: lib/retry_executor.js [hunk 1]
```diff
@@ -37,38 +41,28 @@ class RetryExecutor extends BaseExecutor {
         }
     }
 
-    onMessage(message, onExec) {
+    getOptionIndex(message) {
```
Comment (gwicke) [inline lib/retry_executor.js]: How about something like `shouldProcessMessage`, or `handlerIndex`?

File: lib/base_executor.js [hunk 2]
```diff
@@ -270,6 +284,7 @@ class BaseExecutor {
     }
 
     _catch(message, retryMessage, e) {
+        console.log(e);
```
Comment (d00rman) [inline lib/base_executor.js]: Ups :)

**Comment by d00rman** [comment]
Very nice! One minor nit in-lined. How about bumping the version too since it's a performance improvement?

**Comment by d00rman** [comment]
I'll bump the version and get rid of the `console.log()` in a subsequent PR

**Pull request merged**